numpy/core/meson.build

```diff
@@ -594,9 +585,216 @@ np_core_dep = declare_dependency(
     '.',
     'include',
     'src/common',
-  ]
+  ],
+  dependencies: CPU_DEPENDENCY
 )
 
+# Build dispatch-able sources
+# -----------------------------
+CPU_CONF_DISPATCH_SOURCES = {
+  '_simd': {
+    'dependencies': declare_dependency(
+      sources: [
+        src_file.process('src/_simd/_simd_inc.h.src'),
+        src_file.process('src/_simd/_simd_data.inc.src'),
+      ],
+      include_directories: 'src/_simd/',
+      dependencies: [py_dep, np_core_dep]
+    ),
+    'dispatch': [
+      {
+        'targets': get_option('test-simd'),
+        'sources': src_file.process('src/_simd/_simd.dispatch.c.src'),
+        'config': '_simd.dispatch.h',
+      },
+    ]
+  },
+  '_umath_tests': {
+    'dependencies': [py_dep, np_core_dep],
+    'dispatch': [
+      {
+        'targets': '''
+          avx2 sse41 sse2
+          vsx3 vsx2 vsx
+          asimdhp asimd neon
+          baseline
+        ''',
```

Comment:
> This is probably going to be a little fragile, and seems non-idiomatic.

why? the following code should never fail:
https://github.com/numpy/numpy/blob/8b514f1edf28b01b800b87d4fff7c0c7b34658ea/numpy/core/meson_cpu/meson.build#L32

I just tend to use multi-string to separate the architectures to increase readability but it can be a single string or modify the above code to allow an array-of-str instead.

Reply:
I'd prefer the first version I sketched above for readability, and then `.strip().to_upper().split()` can be left out.

Reply:
It became much simpler now but I had to introduce a new module method to handle multi-targets `mod_features.multi_targets()` rather than counting on sub meson script.
```meson
[
    AVX2, SSE41, SSE2,
    VSX3, VSX2, VSX,
    ASIMDHP, ASIMD, NEON,
]
```

Reply:
I like how this looks now.